src/plugins/expressions/common/execution/types.ts

```diff
@@ -17,6 +17,9 @@
  * under the License.
  */
 
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import type { KibanaRequest } from 'src/core/server';
```

Comment:
I've tested, and so far it seems that as long as we're only importing a type from `server`, it doesn't create issues in the bundle as the typescript is compiled away.